src/content/Institutions/InstitutionView.test.tsx

```diff
@@ -0,0 +1,286 @@
+    userEvent.type(nameInput, "Updated Institution");
+
+    const statusSelect = within(getByTestId("status-select")).getByRole("button");
+    userEvent.click(statusSelect);
+    const listbox = within(getByTestId("status-select")).getByRole("listbox", { hidden: true });
+    userEvent.click(within(listbox).getByText("Inactive"));
+
+    userEvent.click(getByTestId("save-button"));
+
+    await waitFor(() => {
+      expect(global.mockEnqueue).toHaveBeenCalledWith("Institution updated successfully.", {
+        variant: "success",
+      });
+      expect(mockNavigate).toHaveBeenCalledWith("/institutions");
+    });
+  });
+
+  it("navigates back to manage institutions page when cancel button is clicked", () => {
+    const { getByTestId } = render(
+      <TestParent>
+        <InstitutionView _id="new" />
+      </TestParent>
+    );
+
+    const cancelButton = getByTestId("cancel-button");
+    userEvent.click(cancelButton);
+    expect(mockNavigate).toHaveBeenCalledWith("/institutions");
+  });
+
+  it("handles API errors gracefully when creating a new institution", async () => {
+    const { getByTestId } = render(
+      <TestParent mocks={[createInstitutionErrorMock]}>
+        <InstitutionView _id="new" />
+      </TestParent>
+    );
+
+    const nameInput = getByTestId("institutionName-input") as HTMLInputElement;
+    const saveButton = getByTestId("save-button");
+
+    userEvent.type(nameInput, "New Institution");
+    userEvent.click(saveButton);
+
+    await waitFor(() => {
+      expect(global.mockEnqueue).toHaveBeenCalledWith(
+        "There was a problem while creating a new Institution.",
+        {
+          variant: "error",
+        }
+      );
+      expect(mockNavigate).toHaveBeenCalledWith("/institutions");
+    });
+  });
+
+  it("handles API errors gracefully when updating an existing institution", async () => {
+    const { getByTestId } = render(
+      <TestParent mocks={[getInstitutionMock, updateInstitutionErrorMock]}>
+        <InstitutionView _id="institution-1" />
+      </TestParent>
+    );
+
+    await waitFor(() => {
+      expect((getByTestId("institutionName-input") as HTMLInputElement).value).toBe(
+        "Test Institution"
+      );
+    });
+
+    const nameInput = getByTestId("institutionName-input") as HTMLInputElement;
+    userEvent.clear(nameInput);
+    userEvent.type(nameInput, "Updated Institution");
+
+    const statusSelect = within(getByTestId("status-select")).getByRole("button");
+    userEvent.click(statusSelect);
+    const listbox = within(getByTestId("status-select")).getByRole("listbox", { hidden: true });
+    userEvent.click(within(listbox).getByText("Inactive"));
+
+    userEvent.click(getByTestId("save-button"));
+
+    await waitFor(() => {
+      expect(global.mockEnqueue).toHaveBeenCalledWith(
+        "There was a problem while updating the Institution.",
+        { variant: "error" }
+      );
+      expect(mockNavigate).toHaveBeenCalledWith("/institutions");
```

Comment:
See related annotation in InstitutionView.

Probably need to move this expect out of the waitFor else I think it would wait 5s before completing? (could be wrong).

```suggestion
      expect(mockNavigate).not.toHaveBeenCalled();
```

Reply:
Fixed in [223fe55](https://github.com/CBIIT/crdc-datahub-ui/pull/682/commits/223fe55f438c0b15c42e0b6a847ac7330873ed5a) and [f108e54](https://github.com/CBIIT/crdc-datahub-ui/pull/682/commits/f108e543e5270c514e21095ff51688a6f6509d57).